presto-main/src/main/java/com/facebook/presto/execution/ClusterSizeMonitor.java

```diff
@@ -125,15 +140,50 @@ public void stop()
         // remove future if finished (e.g., canceled, timed out)
         future.addListener(() -> {
             timeoutTask.cancel(true);
-            removeFuture(future);
+            removeWorkerFuture(future);
         }, executor);
 
         return future;
     }
 
-    private synchronized void removeFuture(SettableFuture<?> future)
+    public synchronized ListenableFuture<?> waitForMinimumCoordinators()
```

Comment:
instead of having 2 different functions - do you feel we should have a single function `waitForMinimumNodes()` (or some better name) which takes into account both workers and coordinators? Also, this PR does not use this function yet, will this be followed up in a separate PR? If yes, I am curious why?

Reply:
I would prefer to wait for them separately so we could have an accurate message indicating the resource problem (insufficient coordinators/insufficient workers).

You can see how it fits together in the `Extract from` reference in the PR description.  I'm putting these out as small PRs to make it easy to review thoroughly.